### PR TITLE
Configurable produce request version

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -108,4 +108,6 @@
   * Support magic v2 batch/message format (message headers).
   * Use rebar3 as primary build tool for travis-ci, erlang.mk is still supported.
   * Support `SCRAM-SHA-256` and `SCRAM-SHA-512` SASL authentication mechanisms.
+* 3.?.?
```
Comment (viktoriafordos) [inline changelog.md]: Update version number.

Comment (zmstone) [reply changelog.md]: not planing to release it now.
will update when we decide to release.

File: src/brod_producer.erl [hunk 2]
```diff
@@ -465,6 +477,16 @@ maybe_produce(#state{ buffer = Buffer0
       schedule_retry(State#state{buffer = Buffer})
   end.
 
+%% Resolve produce API version to use.
+%% If api version is configured by user, always use configured vresion,
```
Comment (k32) [inline src/brod_producer.erl]: s/vresion/version/

Comment (zmstone) [reply src/brod_producer.erl]: fixed and squashed.

**Review by k32 — APPROVED** [review]

**Review by k32 — APPROVED** [review]

**Review by mikpe — APPROVED** [review]

**Pull request merged**